bin/tests/integration/server_harness/mod.rs

```diff
@@ -46,23 +53,29 @@ impl SocketPorts {
     }
 
     /// Assumes there is only one V4 addr for the IP based on the usage in the Server
-    pub fn get_v4(&self, protocol: Protocol) -> Option<u16> {
+    pub fn get_v4<T: Into<ServerProtocol>>(&self, protocol: T) -> Option<u16> {
         self.0
-            .get(&protocol)
+            .get(&protocol.into())
             .iter()
             .find_map(|ports| if ports.v4 == 0 { None } else { Some(ports.v4) })
     }
 
     /// Assumes there is only one V4 addr for the IP based on the usage in the Server
     #[allow(unused)]
-    pub fn get_v6(&self, protocol: Protocol) -> Option<u16> {
+    pub fn get_v6<T: Into<ServerProtocol>>(&self, protocol: T) -> Option<u16> {
         self.0
-            .get(&protocol)
+            .get(&protocol.into())
             .iter()
             .find_map(|ports| if ports.v6 == 0 { None } else { Some(ports.v6) })
     }
 }
 
+impl From<Protocol> for ServerProtocol {
```

Comment:
Nit: keep this directly below the `ServerProtocol` definition.

Reply:
Moved.